### PR TITLE
Automatically limit integration test parallelism

### PR DESCRIPTION
**Pull request by tstromberg**
This addresses CPU starvation issues I consistently ran into while running CNI tests.

* Go normally runs 1 test per CPU core.
* Each minikube cluster can consume up to 2 CPU cores by default
* minikube binary and integration tests have additional modest overhead
* Current default can cause extreme CPU contention for highly parallel subtests, resulting in timeouts
* This PR effective limits test parallelism to ~1.75 cores

NOTE: This renames `main.go` to `main_test.go`, so that `TestMain` is actually executed. To make the linter happy with symbol resolution, this meant renaming other similar test helper files as well.

### CODE REVIEW

File: test/integration/main_test.go [hunk 1]
```diff
@@ -51,12 +53,44 @@ const (
...
+func setMaxParallelism() {
+	procs := runtime.GOMAXPROCS(0)
+	pval := flag.Lookup("test.parallel").Value.String()
+	pv, err := strconv.Atoi(pval)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to parse --test.parallel value: %q\n", pval)
+		return
+	}
+
+	if procs != pv {
```
Comment (medyagh) [inline test/integration/main_test.go:74]: was this meant to be if proc < pv {  } ?

Comment (tstromberg) [reply test/integration/main_test.go:74]: This is intentional: If someone tests manually --test-parallel, we preserve the value rather than ignoring them..

File: test/integration/testdata/weavenet.yaml [hunk 2]
```diff
@@ -0,0 +1,255 @@
+apiVersion: v1
```
Comment (medyagh) [inline test/integration/testdata/weavenet.yaml:1]: are these yaml files unrelated changes ?

Comment (tstromberg) [reply test/integration/testdata/weavenet.yaml]: Yes. They have now been removed.

**Comment by medyagh** [comment]
looks good but lets wait for VM based tests

**Comment by tstromberg** [comment]
The sample size is small, but this appears to increase the total integration run times slightly:

* Docker: was ~23 minutes, now ~26 minutes
* None: was ~14 minutes, now ~15 minutes

On the other hand, it does make it possible for me to run the CNI tests locally.

**Comment by medyagh** [comment]
> The sample size is small, but this appears to increase the total integration run times slightly:
> 
> * Docker: was ~23 minutes, now ~26 minutes
> * None: was ~14 minutes, now ~15 minutes
> 
> On the other hand, it does make it possible for me to run the CNI tests locally.

I am okay with that given that I think we will see less CPU related flakes
